.changeset/smooth-cameras-prove.md

```diff
@@ -0,0 +1,5 @@
+---
+"@primer/react": major
+---
+
+Move deprecated components to deprecated folder
```

Comment:
A TODO for later @pksjce, we'll need to put more specific breaking change release notes for the components that have moved in this PR.